lib/improver/nbhood/square_kernel.py

```diff
@@ -45,6 +45,37 @@
 MAX_RADIUS_IN_GRID_CELLS = 500
 
 
+def create_cube_with_halo(cube, halo_size_km):
+    """
+    Pad an input cube with a fixed width halo on all sides.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube on original grid
+        halo_size_km (float):
+            Size of border to pad original grid
+
+    Returns:
+        halo_cube (iris.cube.Cube):
+            New cube defining the halo-padded grid (data set to zero)
+    """
+    halo_size_x, halo_size_y = convert_distance_into_number_of_grid_cells(
+        cube, 1000.*halo_size_km)
+
+    # create padded x- and y- coordinates
```

Comment:
Just as a query, could this be done differently by instead of calling `pad_coord` and then created the halo cube, could we  just use the `pad_cube_with_halo` method from the SquareNeighbourhood plugin instead?